pkg/util/kubernetes/apiserver/controllers.go

```diff
@@ -106,23 +102,17 @@ func StartControllers(ctx ControllerContext) errors.Aggregate {
 }
 
 // startMetadataController starts the informers needed for metadata collection.
-// The synchronization of the informers is handled in this function.
+// The synchronization of the informers is handled by the controller.
 func startMetadataController(ctx ControllerContext, c chan error) {
 	metaController := NewMetadataController(
 		ctx.InformerFactory.Core().V1().Nodes(),
 		ctx.InformerFactory.Core().V1().Endpoints(),
 	)
 	go metaController.Run(ctx.StopCh)
-
-	// Wait for the cache to sync
-	c <- SyncInformers(map[InformerName]cache.SharedInformer{
-		nodesInformer:     ctx.InformerFactory.Core().V1().Nodes().Informer(),
-		endpointsInformer: ctx.InformerFactory.Core().V1().Endpoints().Informer(),
-	})
```

Comment:
For more context this was introduced in https://github.com/DataDog/datadog-agent/pull/5291
It's useful to sync informers for Autodiscovery controllers (service and endpoint checks, in `startEndpointsInformer` and `startServicesInformer`) but for Metadata and Autoscalers the controllers are handling the synchronization properly in their `Run` methods. Also at this point the nodes and hpa/wpa informers are not started yet so the sync always timeouts if we try to sync in here (lasy start with `ctx.InformerFactory.Start(ctx.StopCh)`), as opposed to endpoints and service informers where we call `Core().V1().Services().Informer().Run(ctx.StopCh)` and `Core().V1().Endpoints().Informer().Run(ctx.StopCh)` before calling the `SyncInformers` function